pkg/agent/agent.go

```diff
@@ -397,3 +418,236 @@ func (f *Flows) onInterfaceAdded(iface ifaces.Interface) {
 		return
 	}
 }
+
+// ebpfPacketFetcher abstracts the interface of ebpf.FlowFetcher to allow dependency injection in tests
+type ebpfPacketFetcher interface {
+	io.Closer
+	Register(iface ifaces.Interface) error
+
+	LookupAndDeleteMap() map[int][]*byte
+	ReadPerf() (perf.Record, error)
+}
+
+// PacketsAgent instantiates a new agent, given a configuration.
+func PacketsAgent(cfg *Config) (*Packets, error) {
+	plog.Info("initializing Packets agent")
+
+	// configure informer for new interfaces
+	var informer ifaces.Informer
+	switch cfg.ListenInterfaces {
+	case ListenPoll:
+		plog.WithField("period", cfg.ListenPollPeriod).
+			Debug("listening for new interfaces: use polling")
+		informer = ifaces.NewPoller(cfg.ListenPollPeriod, cfg.BuffersLength)
+	case ListenWatch:
+		plog.Debug("listening for new interfaces: use watching")
+		informer = ifaces.NewWatcher(cfg.BuffersLength)
+	default:
+		plog.WithField("providedValue", cfg.ListenInterfaces).
+			Warn("wrong interface listen method. Using file watcher as default")
+		informer = ifaces.NewWatcher(cfg.BuffersLength)
+	}
+
+	plog.Info("[PANO]acquiring Agent IP")
+	agentIP, err := fetchAgentIP(cfg)
+	if err != nil {
+		return nil, fmt.Errorf("acquiring Agent IP: %w", err)
+	}
+
+	// configure selected exporter
+	packetexportFunc, err := buildPacketExporter(cfg)
+	// Log statements to be removed eventually
```

Comment:
Which log statements do you mean?
Can you remove them now if the work is done. else please remove these comments.
Like one below as well.